tidy3d/components/data/monitor_data.py

```diff
@@ -133,7 +133,136 @@ def symmetry_expanded_copy(self) -> AbstractFieldData:
+        tangential_dims.pop(zero_dims[0])
+
+        return tangential_dims
+
+    @property
+    def _tangential_fields(self) -> Dict[str, DataArray]:
+        """For a 2D monitor data, return a dictionary with only the tangential fields. Raise if any
+        of the tangential field components is missing."""
+        tan_dims = self._tangential_dims
+        field_components = ["E" + dim for dim in tan_dims] + ["H" + dim for dim in tan_dims]
+        for field in field_components:
+            if field not in self.field_components:
+                raise DataError(f"Tangential field component {field} is missing in data.")
+        normal_dim = "xyz"[self.monitor.size.index(0)]
+        return {field: getattr(self, field).squeeze(dim=normal_dim) for field in field_components}
+
+    @cached_property
+    def _plane_grid_boundaries(self) -> Tuple[ArrayLike[float, 1], ArrayLike[float, 1]]:
+        """For a 2D monitor data, return the boundaries of the in-plane grid from the stored field
+        coordinates."""
+        tan_dims = self._tangential_dims
+        tan_fields = self._tangential_fields
+        plane_bounds0 = tan_fields["E" + tan_dims[1]].coords[tan_dims[0]].values
+        plane_bounds1 = tan_fields["E" + tan_dims[0]].coords[tan_dims[1]].values
+        return plane_bounds0, plane_bounds1
+
+    @cached_property
+    def _diff_area(self) -> xr.DataArray:
+        """For a 2D monitor data, return the area of each cell in the plane, for use in numerical
+        integrations."""
+        bounds = self._plane_grid_boundaries
+        sizes = [bs[1:] - bs[:-1] for bs in bounds]
+        return xr.DataArray(np.outer(sizes[0], sizes[1]), dims=self._tangential_dims)
+
+    @property
+    def _centered_tangential_fields(self) -> Dict[str, DataArray]:
+        """For a 2D monitor data, get the tangential E and H fields colocated to the cell centers in
+        the 2D plane grid."""
+
+        # Tangential directions and fields
+        tan_dims = self._tangential_dims
+        tan_fields = self._tangential_fields
+
+        # Plane center coordinates
+        bounds = self._plane_grid_boundaries
+        centers = [(bs[1:] + bs[:-1]) / 2 for bs in bounds]
+
+        # Interpolate tangential field components to cell centers
+        interp_dict = dict(zip(tan_dims, centers))
+        centered_fields = {key: val.interp(**interp_dict) for key, val in tan_fields.items()}
+        return centered_fields
+
+    @property
+    def poynting(self) -> xr.DataArray:
+        """Time-averaged Poynting vector for frequency-domain data associated to a 2D monitor."""
+
+        # Tangential fields are ordered as E1, E2, H1, H2
+        tan_fields = self._centered_tangential_fields
+        field_list = list(tan_fields.values())
+        e_x_h_star = field_list[0] * field_list[3].conj() - field_list[1] * field_list[2].conj()
+        poynting = 0.5 * np.real(e_x_h_star)
+        # cast to xr.DataArray as this is technically no longer a ScalarFieldArray
```

Comment:
could it be if we include the normal axis with a single coordinate? or is it maybe better this way?

Reply:
Actually some of this code was done a while ago when I think `ScalarFieldData` may have been reserved for EM fields only? Or maybe that was my understanding at the time. But yeah I've removed the casting, it works right off the bat and now returns `Union[ScalarFieldData, ScalarFieldTimeData]`